sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosItemProperties.java

```diff
@@ -57,21 +76,23 @@ static Document fromObject(Object cosmosItem) {
         return typedItem;
     }
 
-    static String toJsonString(Object cosmosItem, ObjectMapper objectMapper) {
+    static ByteBuffer serializeJsonToByteBuffer(Object cosmosItem, ObjectMapper objectMapper) {
         if (cosmosItem instanceof CosmosItemProperties) {
-            return ((CosmosItemProperties) cosmosItem).toJson();
+            return ((CosmosItemProperties) cosmosItem).serializeJsonToByteBuffer();
         } else {
             if (cosmosItem instanceof Document) {
-                return ((Document) cosmosItem).toJson();
-            }
-            try {
-                return objectMapper.writeValueAsString(cosmosItem);
-            } catch (IOException e) {
-                throw new IllegalArgumentException("Can't serialize the object into the json string", e);
+                return ((Document) cosmosItem).serializeJsonToByteBuffer();
             }
+
+            return Utils.serializeJsonToByteBuffer(objectMapper, cosmosItem);
         }
     }
 
+    static List<CosmosItemProperties> getFromV2Results(List<Document> results) {
```

Comment:
I don't think this function is needed anymore, We can remove this.

Reply:
removed. thanks